corehq/apps/hqwebapp/middleware.py

```diff
@@ -0,0 +1,22 @@
+from django.middleware.csrf import CsrfViewMiddleware, REASON_NO_CSRF_COOKIE, REASON_BAD_TOKEN
+
+from corehq.util.soft_assert import soft_assert
+from django.conf import settings
+
+
+class HQCsrfViewMiddleWare(CsrfViewMiddleware):
+
+    def _reject(self, request, reason):
+        if settings.CSRF_ALWAYS_OFF and reason in [REASON_NO_CSRF_COOKIE, REASON_BAD_TOKEN]:
+
+            warning = "Request at {url} doesn't contain a csrf token. Letting the request pass through for now. "\
+                      "Check if we are sending csrf_token in the corresponding POST form, if not fix it. "\
+                      "Read more here https://github.com/dimagi/commcare-hq/pull/9227".format(
+                          url=request.path
+                      )
+            _assert = soft_assert(notify_admins=True, exponential_backoff=True)
```

Comment:
Yeah, I was thinking of putting it on staging and not on prod. 

Instead of tracking the failed URLs using emails, we can capture the failure with some unique key (view-name (accessible via middleware) or some form of UID free URL) and store it somewhere (a log file or a key-value db). This would make it easy to get a list of all failures in once place and we won't have to blow up our inboxes.

I am planning to fix as many as I can in next 3 days and hand it off to interrupt with instructions after that or look at the failed list when I am back (I will be offline from Thursday)